File: 3288/3288.go
func lengthOfTheLongestIncreasingPa(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := lengthOfTheLongestIncreasingPa(root.Left)
	r := lengthOfTheLongestIncreasingPa(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 3288/3288.c
int lengthOfTheLongestIncreasingPa(struct TreeNode* root) {
    if (!root) return 0;
    int l = lengthOfTheLongestIncreasingPa(root->left);
    int r = lengthOfTheLongestIncreasingPa(root->right);
    return 1 + (l > r ? l : r);
}